patches/tModLoader/Terraria/NPC.cs.patch

```diff
@@ -746,6 +746,15 @@
  					if (buffTime[i] < time)
  						buffTime[i] = time;
  
+@@ -71162,7 +_,7 @@
+ 				if (NPCID.Sets.NoMultiplayerSmoothingByType[type]) {
+ 					netOffset *= 0f;
+ 				}
+-				else if (NPCID.Sets.NoMultiplayerSmoothingByAI[aiStyle]) {
++				else if (aiStyle >= 0 && NPCID.Sets.NoMultiplayerSmoothingByAI[aiStyle]) {
```

Comment:
Check for upper bounds too. There are zero conventions on how to use aiStyle.

Reply:
done, baffling how they use this `Set` by indexing with ai styles anyway